smalltalksrc/VMMaker/CogSimStackEntry.class.st

```diff
@@ -247,7 +255,7 @@ CogSimStackEntry >> moveToVectorReg: reg [
 { #category : #accessing }
 CogSimStackEntry >> offset [
 	"Answer the value of offset"
-	self assert: type = SSBaseOffset.
+	self assert: (self type == SSSpill or: [self type == SSBaseOffset]).
```

Comment:
Can we put a name to this?

Reply:
we could, but it's only in two places, right? if we have a third one we abstract :)

Reply:
Hahah three is the number... it's ok, only want to know why should be `SSSpill` or `SSBaseOffset` (for someone that doesn't know the possible types as me)

Reply:
Actually, both spill and baseoffset represent memory addresses in the form base+offset.
The only difference I've seen so far is that base+offset points to any potential address (e.g, object+variable offset), while spill is always a spill slot in the frame.

But we could differentiate one from the other using the spilled variable 🤷